packages/calypso-e2e/src/lib/pages/google-login-page.ts

```diff
@@ -0,0 +1,117 @@
+import { Page, Locator } from 'playwright';
+
+const selectors = {
+	// Generic
+	button: ( text: string ) => `button[type="button"]:has-text("${ text }")`,
+
+	emailInput: 'input[type="email"]',
+	passwordInput: 'input[type="password"]',
+	phoneInput: 'input[type="tel"][id="phoneNumberId"]',
+	otpInput: 'input[type="tel"][id="idvAnyPhonePin"],input[type="tel"][id="totpPin"]', // Match on both SMS OTP and TOTP inputs.
+};
+
+/**
+ * Represents the login screens shown by Google.
+ */
+export class GoogleLoginPage {
+	/**
+	 * Construct and instance of the EmailClient.
+	 *
+	 * @param {Page} page Handler for instance of the Google login page.
+	 */
+	constructor( private page: Page ) {}
+
+	/**
+	 * Waits until the target locator is stable.
+	 *
+	 * After clicking the "Next" button on the Google login form,
+	 * sliding animations are played. This can interfere with the
+	 * ability for Playwright to successfully interact with the
+	 * target locator.
+	 */
+	private async waitUntilStable( locator: Locator ) {
+		const elementHandle = await locator.elementHandle();
+		await Promise.all( [
+			locator.waitFor( { state: 'visible' } ),
+			elementHandle?.waitForElementState( 'stable' ),
+		] );
+	}
+
+	/**
+	 * Fills the username field.
+	 *
+	 * Note that to avoid bot detection where possible, the
+	 * method used is `type`, not `fill`.
+	 *
+	 * @param {string} email Email address of the user.
+	 */
+	async enterUsername( email: string ): Promise< void > {
+		const locator = this.page.locator( selectors.emailInput );
+		await locator.waitFor( { state: 'visible' } );
+
+		await locator.type( email, { delay: 50 } );
```

Comment:
The `delay` parameter here is part of efforts to circumvent the security measures imposed by Google; it is thought that pretending to type these values will help.